src/server/force-refresh.integration.test.ts

```diff
@@ -0,0 +1,472 @@
+      refreshToken: "initial-app-refresh-token",
+      idToken: "initial-app-id-token",
+      scope: "openid profile app", // Different scope for clarity
+      expiresAt: initialExpiresAt
+    };
+    const initialSession: SessionData = {
+      user: { sub: DEFAULT.sub },
+      tokenSet: initialTokenSet,
+      internal: { sid: DEFAULT.sid, createdAt: Date.now() }
+    };
+
+    // Mock Auth0Client specific methods
+    vi.spyOn(Auth0Client.prototype as any, "getSession").mockResolvedValue(
+      initialSession
+    );
+    // IMPORTANT: saveToSession should NOT be called in app-router mode
+    const mockSaveToSession = vi.spyOn(
+      Auth0Client.prototype as any,
+      "saveToSession"
+    );
+
+    // --- Execution ---
+    const auth0Client = new Auth0Client({
+      domain: DEFAULT.domain,
+      clientId: DEFAULT.clientId,
+      clientSecret: DEFAULT.clientSecret,
+      appBaseUrl: DEFAULT.appBaseUrl,
+      secret: secret
+    });
+
+    // Use expected values from beforeEach setup
+    const expectedRefreshedAccessToken =
+      "getAccessToken-refreshed-access-token";
+    const expectedRefreshedRefreshToken =
+      "getAccessToken-refreshed-refresh-token";
+    const expectedRefreshedExpiresIn = 7200;
+
+    const result = await auth0Client.getAccessToken({
+      refresh: true
+    });
+
+    // --- Assertions ---
+
+    // 1. Assert the returned access token details
+    expect(result).not.toBeNull();
+    expect(result?.token).toBe(expectedRefreshedAccessToken);
+    expect(result?.expiresAt).toBeGreaterThan(initialExpiresAt);
+    const expectedExpiresAtRough =
+      Math.floor(Date.now() / 1000) + expectedRefreshedExpiresIn;
+    expect(result?.expiresAt).toBeGreaterThanOrEqual(
+      expectedExpiresAtRough - 5
+    );
+    expect(result?.expiresAt).toBeLessThanOrEqual(expectedExpiresAtRough + 5);
+    expect(result?.scope).toBe(initialTokenSet.scope); // Scope remains initial
+
+    // 2. Assert saveToSession WAS called (matches current behavior)
+    expect(mockSaveToSession).toHaveBeenCalledOnce();
+    const savedSession = mockSaveToSession.mock.calls[0][0] as SessionData;
+    expect(savedSession.tokenSet.accessToken).toBe(
+      expectedRefreshedAccessToken
+    );
+    expect(savedSession.tokenSet.refreshToken).toBe(
+      expectedRefreshedRefreshToken
+    );
+    expect(savedSession.tokenSet.expiresAt).toBe(result?.expiresAt);
+    expect(savedSession.tokenSet.scope).toBe(initialTokenSet.scope);
+
+    // 3. Assert mockFetch was called
+    const fetchCalls = mockFetch.mock.calls;
+    expect(fetchCalls.length).toBeGreaterThan(0);
+    const tokenEndpointCall = fetchCalls.find((call: any) => {
+      let urlString: string;
+      if (call[0] instanceof URL) {
+        urlString = call[0].toString();
+      } else if (call[0] instanceof Request) {
+        urlString = call[0].url;
+      } else {
+        urlString = call[0];
+      }
+      return urlString.endsWith("/oauth/token");
+    });
```

Comment:
We do not need to assert this, we know this is the case by asserting the returned values.